### ISSUE TITLE
Would be nice to have a "host.fact.user" fact

### ISSUE THREAD

**Issue by taranlu-houzz** [Fizzadar/pyinfra#483]
**Is your feature request related to a problem? Please describe.**
Right now, if I want to get the user that is being used to connect to a given remote machine I have the following options:

- Add group data that specifies the `user` for a given remote.
- Use `host.data.ssh_user` (assuming an ssh connection).
- Grab the `host.fact.home` fact and then split out the user.
- Write a custom fact that does the above.

When running adhoc commands, the first two methods don't work, and the last two methods require duplicate code in all operations that happen to need the "user," which works but is not ideal.

**Describe the solution you'd like**
It would be great to have a built-in `host.fact.user`. The `host.fact.home` fact is almost what I need, but falls short when it comes to running adhoc commands.

**Comment by Fizzadar** [comment]
I like this idea; to get the current user right? Could be implemented by executing `whoami`, but that would mean calling it with `sudo` would return `root` (or the `sudo_user`), which I think is useful. Ie:

```py
# deploy-no-sudo.py
print(host.fact.user)  # SSH user
```

vs:

```py
# deploy-with-global-sudo.py

SUDO = True
print(host.fact.user)  # root
```

This would be the same behaviour we have the `host.fact.home`. I think to get the SSH user (or winrm user, etc, etc) that is something specific to the connection type, and makes sense to be retrieved from there.

**Comment by taranlu-houzz** [comment]
Yep, I pretty much just need the output of `host.fact.home` split to just the user. `whoami` was also the first thing my mind jumped to, but I wonder if there might be some systems/scenarios where that is not ideal (aside from not working on Windows). Maybe it could try several different methods, possibly starting with checking the `$USER` env var first?

**Comment by Fizzadar** [comment]
Turns out both work everywhere!

```
--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/fedora24] vagrant
[@vagrant/fedora24] vagrant
    [@vagrant/fedora24] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/freebsd12] vagrant
[@vagrant/freebsd12] vagrant
    [@vagrant/freebsd12] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/centos8] vagrant
[@vagrant/centos8] vagrant
    [@vagrant/centos8] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/centos7] vagrant
[@vagrant/centos7] vagrant
    [@vagrant/centos7] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/centos6] vagrant
[@vagrant/centos6] vagrant
    [@vagrant/centos6] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/opensuse_tumbleweed] vagrant
[@vagrant/opensuse_tumbleweed] vagrant
    [@vagrant/opensuse_tumbleweed] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/debian8] vagrant
[@vagrant/debian8] vagrant
    [@vagrant/debian8] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/opensuse_leap15] vagrant
[@vagrant/opensuse_leap15] vagrant
    [@vagrant/opensuse_leap15] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/ubuntu18] vagrant
[@vagrant/ubuntu18] vagrant
    [@vagrant/ubuntu18] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/debian9] vagrant
[@vagrant/debian9] vagrant
    [@vagrant/debian9] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/ubuntu16] vagrant
[@vagrant/ubuntu16] vagrant
    [@vagrant/ubuntu16] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/openbsd6] vagrant
[@vagrant/openbsd6] vagrant
    [@vagrant/openbsd6] Success

--> Starting operation: Server/Shell (echo $USER; whoami)
[@vagrant/ubuntu20] vagrant
[@vagrant/ubuntu20] vagrant
    [@vagrant/ubuntu20] Success
```

I'll go with `$USER` as it matches the `home` fact!

**Comment by Fizzadar** [comment]
I have now implemented the `user` fact in https://github.com/Fizzadar/pyinfra/commit/fdf74a444ed91421fc29771a4a5d5bd06fd797b9.

**Comment by Fizzadar** [comment]
Now released in v1.3!
